### PR TITLE
#182998134-SETUP LATEST REACT ROUTER VERSION(REACT-ROUTER-DOM)

### PR DESCRIPTION
**Pull request by ChrisOnsando**
#### What does this Pull Request do?
- [ ] Install and configure react router
- [ ] Create a login route
- [ ] Test the login routes
- [ ] Implement a project  folder structure
#### Description of Task to be completed?
- [ ] set up latest react router DOM version 6
#### How should this be manually tested?
- [ ] run `npm start`
#### Any background context you want to provide?
- [ ] React router enables the navigation among views of various components in the React Application, allows changing the browser URL, and keeps the UI in sync with the URL
#### What are the relevant pivotal tracker story?
- [ ] (182998134)-https://www.pivotaltracker.com/story/show/182998134
#### Screenshots (if appropriate)
Screenshot from 2022-08-17 15-41-17
Screenshot from 2022-08-18 09-30-39
#### Questions:
- [ ] N/A

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -1,18 +1,17 @@
...
+import React from 'react';
+import { Routes, Route } from 'react-router-dom';
+import Home from './components/Home';
+import Login from './components/Login';
+import Navbar from './components/Navbar';
 
 function App() {
   return (
     <div className="App">
-      <header className="App-header">
```
Comment (narravabrion) [inline src/App.jsx:7]: Delete classes that you haven't written yourself

Comment (ChrisOnsando) [reply src/App.jsx:7]: Alright

Comment (ChrisOnsando) [reply src/App.jsx:7]: The classes are not there

File: src/components/Login.jsx [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+
+function Login() {
+  return (
+    <div className="login">
+      <h1>Login to the project</h1>
```
Comment (dalienst) [inline src/components/Login.jsx]: Login to the application

File: src/App.jsx (hunk 1 above)
Comment (ChrisOnsando) [reply src/App.jsx:7]: The initial classes are not there

**Review by dalienst — COMMENTED** [review]
I have reviewed the code

File: src/components/Login.jsx (hunk 2 above)
Comment (ChrisOnsando) [reply src/components/Login.jsx]: Alright

File: src/index.jsx [hunk 3]
```diff
@@ -1,14 +1,15 @@
...
 import ReactDOM from 'react-dom/client';
+import { BrowserRouter } from 'react-router-dom';
 import './index.css';
 import App from './App';
 import reportWebVitals from './reportWebVitals';
 
 const root = ReactDOM.createRoot(document.getElementById('root'));
 root.render(
-  <React.StrictMode>
+  <BrowserRouter>
```
Comment (KaigaMunene) [inline src/index.jsx]: You shouldn't have changed this

**Review by KaigaMunene — COMMENTED** [review]
- The routes should be in there own file 
- check on lazy loading in react

File: src/index.jsx (hunk 3 above)
Comment (ChrisOnsando) [reply src/index.jsx]: Why? how will routes work with StrictMode

Comment (KaigaMunene) [reply src/index.jsx]: React.Scrictmode won't affect the routes from working

File: src/constants/Links.jsx [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+        </li>
+        <li>
+          <NavLink to="/login">Login</NavLink>
+        </li>
+      </ul>
+    </div>
+  );
+}
+
+export default PublicLinks;
```
Comment (KaigaMunene) [inline src/constants/Links.jsx]: research on a better way to do this, you just copy pasted the navbar component

Comment (ChrisOnsando) [reply src/constants/Links.jsx]: Okay

File: src/App.jsx [hunk 5]
```diff
@@ -1,18 +1,10 @@
...
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a className="App-link" href="https://reactjs.org">
-          Learn React
-        </a>
-      </header>
+      <BrowserRoutes />
```
Comment (mainamuriuki274) [inline src/App.jsx]: Told you to change the name to Routes

File: src/constants/BrowserRoutes.jsx [hunk 6]
```diff
@@ -0,0 +1,23 @@
+import React, { Suspense } from 'react';
+import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
+import PublicLinks from './Links';
+
+const Home = React.lazy(() => import('../pages/Home'));
+const Login = React.lazy(() => import('../pages/Login'));
+
+function BrowserRoutes() {
```
Comment (mainamuriuki274) [inline src/constants/BrowserRoutes.jsx]: Rename file and function to Routes

File: src/constants/BrowserRoutes.jsx [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+  return (
+    <div className="App">
+      <Router>
+        <Suspense fallback={<div>Loading...</div>}>
+          <Routes>
+            <Route exact path={PublicLinks.home} element={<Home />} />
+            <Route path={PublicLinks.Login} element={<Login />} />
```
Comment (mainamuriuki274) [inline src/constants/BrowserRoutes.jsx]: check on correct case for JS, should be camelCase

File: src/constants/BrowserRoutes.jsx [hunk 8]
```diff
@@ -0,0 +1,23 @@
+import React, { Suspense } from 'react';
+import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
+import PublicLinks from './Links';
+
+const Home = React.lazy(() => import('../pages/Home'));
+const Login = React.lazy(() => import('../pages/Login'));
+
+function BrowserRoutes() {
+  return (
+    <div className="App">
```
Comment (mainamuriuki274) [inline src/constants/BrowserRoutes.jsx]: no need to wrap in a div

File: src/constants/Links.jsx [hunk 9]
```diff
@@ -0,0 +1,6 @@
+const PublicLinks = {
```
Comment (mainamuriuki274) [inline src/constants/Links.jsx]: PublicLinks should be in camelCase

File: src/features/layout/NavBar.jsx [hunk 10]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import { NavLink } from 'react-router-dom';
+import PublicLinks from '../../constants/Links';
+
+function Navbar() {
```
Comment (mainamuriuki274) [inline src/features/layout/NavBar.jsx]: should be under layouts, not under features layouts is another folder in src

**Review by mainamuriuki274 — CHANGES_REQUESTED** [review]

File: src/features/layout/NavBar.jsx (hunk 10 above)
Comment (ChrisOnsando) [reply src/features/layout/NavBar.jsx]: Okay

**Review by mainamuriuki274 — APPROVED** [review]

**Pull request merged**